core/deque_test.go

```diff
@@ -0,0 +1,139 @@
+package core_test
+
+import (
+	"fmt"
+	"math/rand"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/dicedb/dice/core"
+	"gotest.tools/v3/assert"
+)
+
+var randGenerator *rand.Rand
+
+func init() {
```

Comment:
Please avoid common init methods since they will automatically be called for all other tests in this package. Instead name it initDeque or something similar and call it at the start of each test function.